lib/gradebook.rb

```diff
@@ -1 +1,27 @@
 
+class GradeBook
+  attr_reader :instructor
+              :courses
+
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+    
+  end
+
+  def courses
+    @courses
+  end
+
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    students = {}
```

Comment:
I think this method is pretty close to working and could be fixed by giving this new hash default values:
```suggestion
    students = Hash.new([])
```